crates/client/rpc/src/errors.rs

```diff
@@ -42,6 +45,15 @@ pub enum StarknetRpcApiError {
     UnimplementedMethod = 501,
 }
 
+impl From<DispatchError> for StarknetRpcApiError {
+    fn from(err: DispatchError) -> Self {
+        match err {
+            _ if err == PalletError::<Runtime>::ContractNotFound.into() => StarknetRpcApiError::ContractNotFound,
+            _ => StarknetRpcApiError::ContractError,
```

Comment:
Just do a if/esle, it will be more idiomatic, there is no reason to prefer match here